File: sdk/nodejs/src/resources/cloud_vm.js
"use strict";

const { BaseResource } = require("./base");

class CloudVmResource extends BaseResource {
  async getUpgrade(serviceId) {
    return this._http.get("/api/service/{id}/upgrade");
  }

  async upgrade(serviceId, data) {
    return this._http.post("/api/service/{id}/upgrade", data);
  }

  async getResources(serviceId) {
    return this._http.get("/api/service/{id}/resources");
  }

  async getTemplates(serviceId, vmId) {
    return this._http.get("/api/service/{id}/templates/{vmid}");
  }

  async getMemoryUsage(serviceId, vmId) {
    return this._http.get("/api/service/{id}/vms/{vmid}/usage/memory");
  }

  async listVmImages(serviceId, vmId) {
    return this._http.get("/api/service/{id}/vms/{vmid}/images");
  }

  async createVmImage(serviceId, vmId, data) {
    return this._http.post("/api/service/{id}/vms/{vmid}/images", data);
  }

  async setBootOrder(serviceId, vmId, data) {
    return this._http.post("/api/service/{id}/vms/{vmid}/boot", data);
  }

  async toggleTunTap(serviceId, vmId, data) {
    return this._http.post("/api/service/{id}/vms/{vmid}/tuntap", data);
  }

}

module.exports = { CloudVmResource };

File: sdk/csharp/Resources/CloudVm.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class CloudVm : BaseResource
{
    public CloudVm(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> UpgradeAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/upgrade", id));
    }

    public async Task<Dictionary<string, object?>> DoUpgradeAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/upgrade", id), data);
    }

    public async Task<Dictionary<string, object?>> ResourcesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/resources", id));
    }

    public async Task<Dictionary<string, object?>> TemplateAsync(int id, int vmid)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/templates/{vmid}", id, vmid));
    }

    public async Task<Dictionary<string, object?>> MemoryUsageAsync(int id, int vmid)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/{vmid}/usage/memory", id, vmid));
    }

    public async Task<Dictionary<string, object?>> VmImagesAsync(int id, int vmid)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/{vmid}/images", id, vmid));
    }

    public async Task<Dictionary<string, object?>> CreateVmImageAsync(int id, int vmid, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/images", id, vmid), data);
    }

    public async Task<Dictionary<string, object?>> BootOrderAsync(int id, int vmid, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/boot", id, vmid), data);
    }

    public async Task<Dictionary<string, object?>> TuntapAsync(int id, int vmid, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/tuntap", id, vmid), data);
    }
}
